### PR TITLE
Update link for BPF design Q&A

### PR DESCRIPTION
**Pull request by li-ch**
Previous link does not work any more.

### CODE REVIEW

**Comment by qmonnet** [comment]
Thanks!

The .txt file [has been removed from the kernel tree](https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/commit/?id=192092faa02dd5e5d1ff875d7512a5d803db95a0), are you sure that it will remain accessible at this URL? It might be worth to switch to the RST file anyway, which is likely the only one susceptible to receive updates in the future.

**Comment by li-ch** [comment]
Sure, it makes sense. I've updated it to the rst version.

File: readme.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ to start with “kernel documentation”, which is dense).
 -   [linux/Documentation/networking/filter.txt](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/filter.txt).
     Kernel documentation: eBPF specification (somewhat outdated; information
     should still be valid, but not exhaustive).
--   [linux/Documentation/bpf/bpf_design_QA.txt](https://www.kernel.org/doc/Documentation/bpf/bpf_design_QA.txt).
+-   [linux/Documentation/bpf/bpf_design_QA.txt](https://www.kernel.org/doc/Documentation/bpf/bpf_design_QA.rst).
```
Comment (qmonnet) [inline readme.md]: Great, please also update the text of the link (it still says `linux/Documentation/bpf/bpf_design_QA.txt`, should end with `.rst` instead.

**Comment by qmonnet** [comment]
All good, I merged it. Thanks a lot!

**Pull request merged**